### ISSUE TITLE
How do I do multiple graphs training?

### ISSUE THREAD

**Issue by HYDesmondLiu** [pyg-team/pytorch_geometric#824]
## ❓ Questions & Help
Is it possible to do multiple graphs training with outputs of all nodes classification? If so, is there a sample code or any instruction?
    Thanks very much.

**Comment by rusty1s** [comment]
Sure, that‘s possible, just input your graphs into the DataLoader and train against your node targets. There is no example, but it is quite similar to gcn.py except that you need to iterate over batches.

**Comment by HYDesmondLiu** [comment]
So I just stack all my feature vectors up into one array?

**Comment by rusty1s** [comment]
Yes, this is exactly what the `DataLodaer` is doing internally.

**Comment by jaredsagendorf** [comment]
How does this work when you only want to train using a subset of the nodes (i.e. a semi-supervised fashion)?

If I combine several graphs into a batch using the DataLoader, and say each graph has a subset of node with labels that I want to use to train, how do I index these nodes in the batch?

**Comment by rusty1s** [comment]
I suggest that you add a `mask` tensor of size `[num_nodes]` to each `Data` object. You can then filter out the correct nodes by running `x[mask]` even in a mini-batch scenario.

**Comment by Fr33k0** [comment]
Hi, I have similar question. 
I want to train the model with multiple graphs and output all nodes classes for each graph and then use this model for all node classification in unseen graphs. I stored all my training graphs in list of `dataset = [Data(...), Data(...)]` and then input this list to `DataLoader(dataset, batch_size=8)`. I wanted to start with your `gcn.py` but (if I correctly understand) the pooling operation changes shape of graph, so should I remove it from `forward()` function? Because if I input graph with shape `[num_nodes, num_features] = [100, 20]` to model my BCELoss function expects `[num_nodes, 1] = [100, 1]` shape. This type of classification doesn't support the pooling operation or am I wrong?

**Comment by rusty1s** [comment]
Yes, you do not need a pooling operation for node classification (there's no pooling operation in `gcn.py`). For an example of doing node classification on multiple graphs, see `examples/ppi.py`.

**Comment by Fr33k0** [comment]
Actually I was looking at `benchmark/kernel/gcn.py`. Thanks for fast reply!
